runtime/make/Makefile.runtime.hwloc-system

```diff
@@ -0,0 +1,18 @@
+# Copyright 2004-2017 Cray Inc.
+# Other additional copyright holders may be indicated within.
+# 
+# The entirety of this work is licensed under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+# 
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+RUNTIME_DEFS += -DCHPL_HAS_HWLOC
```

Comment:
Can we replace the

```c
#ifdef CHPL_HAS_HWLOC
```

with 

```c
#include "chpl-env-gen.h"
#if !defined(CHPL_HWLOC_NONE)
```

or 

```c
#include "chpl-env-gen.h"
#if defined(CHPL_HWLOC_HWLOC) || defined(CHPL_HWLOC_SYSTEM)
```

and remove the need to set `-DCHPL_HAS_HWLOC`?